File: ws/pilatasig/u2/WS_15_Inheritances/ZooApp/src/main/java/ec/edu/espe/zooapp/model/Mammal.java

package ec.edu.espe.zooapp.model;

import java.util.ArrayList;
import java.util.Date;

/**
 *
 * @author David Pilatasig
 */
public class Mammal extends Animal {
    private float maxHairSize;
    private int mammaryGlandNumber;
    
    public void breastFeed(){
        System.out.println("Breastfeeding the children ");
    }

    public float getMaxHairSize() {
        return maxHairSize;
    }

    public void setMaxHairSize(float maxHairSize) {
        this.maxHairSize = maxHairSize;
    }

    public int getMammaryGlandNumber() {
        return mammaryGlandNumber;
    }

    public void setMammaryGlandNumber(int mammaryGlandNumber) {
        this.mammaryGlandNumber = mammaryGlandNumber;
    }

    public Mammal(float maxHairSize, int mammaryGlandNumber, int id, String name, Date bornOnDate, float weight, char sex, ArrayList<Vaccine> vaccines) {
        super(id, name, bornOnDate, weight, sex, vaccines);
        this.maxHairSize = maxHairSize;
        this.mammaryGlandNumber = mammaryGlandNumber;
    }

    @Override
    public String toString() {
        return "Mammal{" + super.toString()+ ", maxHairSize=" + maxHairSize + ", mammaryGlandNumber=" + mammaryGlandNumber + '}';
    }
    
    
    
}

File: hw/pilatasig/u2/ZooApp JavaScript/ec/edu/espe/zoo/model/Mammal.js
import { Animal } from "./Animal.js";

export class Mammal extends Animal {
    constructor(maxHairSize, mammaryGlandNumber, id, name, bornOnDate, weight, sex, vaccines = []) {
        super(id, name, bornOnDate, weight, sex, vaccines); // Llamada al constructor de la clase base (Animal)
        this._maxHairSize = maxHairSize; // float
        this._mammaryGlandNumber = mammaryGlandNumber; // int
    }

   
    breastFeed() {
        console.log("Breastfeeding the children");
    }

    
    get maxHairSize() {
        return this._maxHairSize;
    }

        set maxHairSize(value) {
        this._maxHairSize = value;
    }

    r
    get mammaryGlandNumber() {
        return this._mammaryGlandNumber;
    }

    
    set mammaryGlandNumber(value) {
        this._mammaryGlandNumber = value;
    }

    // Método toString
    toString() {
        return `Mammal{${super.toString()}, maxHairSize=${this._maxHairSize}, mammaryGlandNumber=${this._mammaryGlandNumber}}`;
    }
}
